kiwi_lint/similar_string.py

```diff
@@ -0,0 +1,146 @@
+                    )
+
+    @staticmethod
+    def clean_string(text):
+        """
+        This method removes the operators and other punctuations
+        used in the string to avoid unwanted data to add up to
+        the similarity between the strings.
+        """
+        cleaned = ""
+        for char in text:
+            if char not in string.punctuation:
+                cleaned += char
+        return cleaned
+
+    def check_similar_string(self, translation_string):
+        cleaned_translation_string = self.clean_string(translation_string)
+        for key in self._dict_of_strings:
+            if (
+                levenshtein.normalized_similarity(
+                    cleaned_translation_string, self.clean_string(key)
+                )
+                >= self.threshold
+            ):
+                return key
+        return None
+
+    @utils.check_messages("similar-string")
+    def visit_call(self, node):
+        if not (
+            isinstance(node.func, astroid.Name)
+            and node.func.name in ("_", "gettext_lazy")
+        ):
+            return
+        if not isinstance(node.args[0], astroid.nodes.Const):
+            return
+
+        translation_string = node.args[0].value
+
+        if translation_string in self._dict_of_strings:
+            return
+
+        similar_string = self.check_similar_string(translation_string)
+
+        if similar_string:
+            self.add_message(
+                "similar-string",
+                node=node,
+                args=(
+                    translation_string,
+                    similar_string,
+                ),
+            )
+            return
+
+        self._dict_of_strings[translation_string] = True
+
+    def parse_translation_string(self, filename, lines):
+        for lineno, line in enumerate(lines):
+            match_in_line = re.finditer(r'{% trans "([\w ]+)" %}', line)
+            for match in match_in_line:
+                if filename not in self.error_messages:
+                    self.error_messages[filename] = []
+
+                translation_string = match.group(1)
+
+                if translation_string in self._dict_of_strings:
+                    continue
+
+                similar_string = self.check_similar_string(translation_string)
+
+                if similar_string:
+                    self.error_messages[filename].append(
+                        (
+                            lineno + 1,
+                            match.start() + 10,
+                            translation_string,
+                            similar_string,
+                        )
+                    )
+                    continue
+
+                self._dict_of_strings[translation_string] = True
+
+    def close(self):
+        for filepath in self.all_html_files:
+            with open(filepath, "r") as file:
+                read_lines = file.readlines()
+                file_lines = tuple(read_lines)
```

Comment:
the conversion to tuple doesn't seem to be necessary and the intermeiate variables either. You can pass `file.readlines()` directly to `parse_translation_string()`